File: WeNeed1/WeNeed1.Service/Database/Comment.cs
using System;
using System.Collections.Generic;

namespace WeNeed1.Service.Database;

public partial class Comment
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string? Content { get; set; }

    public DateTime Created { get; set; }

    public int MatchId { get; set; }

    public virtual Match Match { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: WeNeed1/MobileUI/playermobile/lib/models/comment.dart
class Comment {
  final int id;
  final String? content;
  final DateTime? created;
  final int? matchId;
  final int? userId;
  final String? userName;
  final String? profilePicture;

  Comment({
     required this.id,
     this.content,
     this.created,
     this.matchId,
     this.userId,
     this.userName,
    this.profilePicture,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    final user = json['user'] ?? {};
    return Comment(
      id: json['id'],
      content: json['content'],
      created: DateTime.parse(json['created']),
      matchId: json['matchId'],
      userId: json['userId'],
      userName: user['userName'] ?? '',
      profilePicture: user['profilePicture'],
    );
  }
}
